components/raftstore/src/store/fsm/store.rs

```diff
@@ -2525,6 +2528,44 @@ impl<'a, EK: KvEngine, ER: RaftEngine, T: Transport> StoreFsmDelegate<'a, EK, ER
         self.register_compact_lock_cf_tick();
     }
 
+    fn on_wake_up_regions(&self, abnormal_stores: Vec<u64>) {
+        info!("try to wake up all hibernated regions in this store";
+            "to_all" => abnormal_stores.is_empty());
+        let meta = self.ctx.store_meta.lock().unwrap();
+        for region_id in meta.regions.keys() {
+            let region = &meta.regions[region_id];
+            if !region_on_stores(region, &abnormal_stores) {
```

Comment:
Is it a N * N loop? (assume we have N regions in this store)
If so, will it be too slow?

Reply:
On the one hand, more specifically, it's a `N * K` loop (assume we have N regions in this store, and we have K abnormal stores, K << N), so it's might be acceptable.

And on the other hand, I wanna give less pressure to PD, so the selection strategy of abnormal regions by `region_on_stores`, is pushed down to TiKV side. In this way, we just pass `abnormal stores` rather than `regions` to other TiKV nodes and let TiKV do the related computation in a `N * K` loop.

Reply:
It's K * N * 3, fairly good, close

Reply:
It's K * N * 3, fairly good